src/script_opt/ZAM/Compile.h

```diff
@@ -53,7 +52,7 @@ class OpaqueVals {
 
 class ZAMCompiler {
 public:
-    ZAMCompiler(ScriptFunc* f, std::shared_ptr<ProfileFunc> pf, ScopePtr scope, StmtPtr body,
+    ZAMCompiler(ScriptFunc* f, ProfileFuncs& pfs, std::shared_ptr<ProfileFunc> pf, ScopePtr scope, StmtPtr body,
```

Comment:
Nothing to do, because this is down in the weeds and not an API for someone to use.  Mostly amusing how this is using a raw pointer, a reference, a few std::shared_ptrs and then an IntrusivePtrs, too.

I guess `ProfileFuncs` could just be a raw pointer, it's just as unsafe as storing the reference, no?

Reply:
Oof, yeah it is unnecessarily cluttered. I've noted this for potential streamlining when I do this month's script optimization (which I've put on hold until this PR is resolved & merged, given the number of changes it introduces).